src/Margay.cpp

```diff
@@ -800,30 +817,60 @@ void Margay::resetWDT()  //Send a pulse to "feed" the watchdog timer
 	digitalWrite(WDHold, LOW);
 }
 
-void Margay::externalI2C(bool state)
+void Margay::switchExternalI2C(bool desiredState)
 {
 	/*
 	Must be ON to read off-board sensors, OFF to read on-board sensors and RTC
 	Serves to isolate these s.t. I2C addresses may not clash.
 	*/
 	pinMode(I2C_SW, OUTPUT);
-	if ( state == ON ){
+
+	if ( desiredState == ON ) {
 		digitalWrite(I2C_SW, HIGH);
+		externalI2COn = digitalRead(I2C_SW);
 	}
-	else{
+	else {
 		digitalWrite(I2C_SW, LOW);
+		externalI2COn = digitalRead(I2C_SW);
 	}
 	delay(1); // Any time needed to switch states; may not be necessary
 }
 
+void Margay::closeFarmGateI2C(bool initStateI2C)
```

Comment:
Can open or close gate, so rename:
```suggestion
void Margay::farmGateI2C(bool initStateI2C)
```